### PR TITLE
Add EIP: Increase the MAX_EFFECTIVE_BALANCE

### PR DESCRIPTION
**Pull request by michaelneuder**
EIP PR for https://ethereum-magicians.org/t/eip-7251-increase-the-max-effective-balance/15982

See https://gist.github.com/michaelneuder/cafabcfcfcccc45e44ab9d6b1c7b4e1d for related documents

### CODE REVIEW

File: EIPS/eip-increase-maxeb.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+```
+
+## Rationale
+
+See https://notes.ethereum.org/@mikeneuder/maxeb-status for current status and
+design goals.
+
+### Features
+
+1. **`0x02` withdrawal credential signals compounding.** These validators will be excluded from the partial withdrawals sweep and their balance compounds.
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply EIPS/eip-increase-maxeb.md:303]: Ah, I see this is still "in discussion" below even though 7002 is noted as a dependency  above. I would note it as a point of discussion in that line item

**Review by djrtwo — COMMENTED** [review]
Left a few quick comments but haven't done a deep review of the granular changes

File: EIPS/eip-increase-maxeb.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+The large and rapidly growing validator set puts an [unnecessary burden](https://ethresear.ch/t/removing-unnecessary-stress-from-ethereums-p2p-network/15547)
+on the p2p network and blocks critical protocol upgrades such as [SSF](https://ethresear.ch/t/a-simple-single-slot-finality-protocol/14920). See [a modest proposal](https://ethresear.ch/t/increase-the-max-effective-balance-a-modest-proposal/15801) for more context.
+
+## Specification
+
+***Consensus layer changes***
+
+See https://github.com/michaelneuder/consensus-specs/pull/3/files for a diff
+view of the CL spec changes. Changes are also copied below for convenience.
```
[the remark this answers is not in the document]
Comment (michaelneuder) [reply EIPS/eip-increase-maxeb.md:36]: great :-) that was annoying to put together so im happy we don't have to maintain it lol.

File: EIPS/eip-increase-maxeb.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelneuder) [reply EIPS/eip-increase-maxeb.md:303]: done!

File: EIPS/eip-increase-maxeb.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+---
```
Comment (abcoathup) [inline EIPS/eip-increase-maxeb.md:1]:
```suggestion
---
eip: 7251
```
Issuing EIP number using PR number.  Please also update the filename.

File: EIPS/eip-increase-maxeb.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+6. Modify `initiate_validator_exit` to rate limit the exit queue by balance rather than number of vaidators.
+7. Modify `initialize_beacon_state_from_eth1` to use `MIN_ACTIVATION_BALANCE`.
+8. Add `process_pending_balance_deposits` to epoch processing logic.
+9. Modify `process_registry_updates` to activate all eligible validators.
+10. Modify `get_validator_from_deposit` to initialize effective balance to zero (its updated by the pending deposit flow).
+11. Modify `apply_deposit` to store incoming deposits in `state.pending_balance_deposits`. 
+12. Modify `is_aggregator` to be weight-based.
+13. Modify `compute_weak_subjectivity_period` to use new churn limit function.
+14. Add `has_compounding_withdrawal_credential` to check for `0x02` credential.
+15. Modify `is_fully_withdrawalable_validator` to check for compounding credentials.
```
Comment (ralexstokes) [inline EIPS/eip-increase-maxeb.md:64]:
```suggestion
15. Modify `is_fully_withdrawable_validator` to check for compounding credentials.
```

File: EIPS/eip-increase-maxeb.md (hunk 3 above)
Comment (michaelneuder) [reply EIPS/eip-increase-maxeb.md]: done!

File: EIPS/eip-increase-maxeb.md (hunk 4 above)
Comment (michaelneuder) [reply EIPS/eip-increase-maxeb.md]: done!

File: EIPS/eip-7251.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+11. Modify `apply_deposit` to store incoming deposits in `state.pending_balance_deposits`. 
+12. Modify `is_aggregator` to be weight-based.
+13. Modify `compute_weak_subjectivity_period` to use the new churn limit function.
+14. Add `has_compounding_withdrawal_credential` to check for the `0x02` credential.
+15. Modify `is_fully_withdrawable_validator` to check for compounding credentials.
+16. Add `get_validator_excess_balance` to calculate the excess balance of validators.
+17. Modify `is_partially_withdrawable_validator` to check for excess balance.
+18. Modify `get_expected_withdrawals` to use excess balance.
+
+
```
Comment (lightclient) [inline EIPS/eip-7251.md:69]:
```suggestion
```

Comment (lightclient) [inline EIPS/eip-7251.md:69]:
```suggestion
```

**Review by lightclient — APPROVED** [review]

File: EIPS/eip-7251.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+10. Modify `get_validator_from_deposit` to initialize the effective balance to zero (it's updated by the pending deposit flow).
+11. Modify `apply_deposit` to store incoming deposits in `state.pending_balance_deposits`. 
+12. Modify `is_aggregator` to be weight-based.
+13. Modify `compute_weak_subjectivity_period` to use the new churn limit function.
+14. Add `has_compounding_withdrawal_credential` to check for the `0x02` credential.
+15. Modify `is_fully_withdrawable_validator` to check for compounding credentials.
+16. Add `get_validator_excess_balance` to calculate the excess balance of validators.
+17. Modify `is_partially_withdrawable_validator` to check for excess balance.
+18. Modify `get_expected_withdrawals` to use excess balance.
+
```
Comment (lightclient) [inline EIPS/eip-7251.md:68]:
```suggestion
```

Comment (michaelneuder) [reply EIPS/eip-7251.md:68]: done!

**Pull request merged**